Bottom: 4e6aa4a637717438e96073db4601629e7139185a
Top:    62f4d4743f20ccb238b49247fc21798a98005f55
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-19 10:05:17 +0800

Refresh of autofs-5.0.7-fix-bad-mkdir-permission-on-create.patch

---

diff --git a/daemon/automount.c b/daemon/automount.c
index 6b55103..a22de2d 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -122,7 +122,7 @@ static int do_mkdir(const char *parent, const char *path, mode_t mode)
 		status = statfs(parent, &fs);
 	if ((status != -1 && fs.f_type == (__SWORD_TYPE) AUTOFS_SUPER_MAGIC) ||
 	    contained_in_local_fs(path)) {
-		int mask = umask(0022);
+		mode_t mask = umask(0022);
 		int ret = mkdir(path, mode);
 		void umask(mask);
 		if (ret == -1) {
